File: arrays_and_hashing/217.py
# https://leetcode.com/problems/contains-duplicate/
# O(n), O(n)


from typing import List


class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        collect = dict()
        for num in nums:
            if num in collect:
                collect[num] += 1
                if collect[num] >= 2:
                    return True
            else:
                collect[num] = 1
        return False

File: arrays_and_hashing/217.cpp
#include <iostream>
#include <vector>
#include <unordered_set>
using namespace std;

// https://leetcode.com/problems/contains-duplicate/
// O(n), O(n)
class Solution {
public:
  bool containsDuplicate(vector<int> &nums) {
    unordered_set<int> collect;
    for (int num : nums) {
      if (collect.find(num) != collect.end()) {
        return true;
      } else {
        collect.insert(num);
      }
    }
    return false;
  }
};

int main(int argc, char *argv[]) {
  vector<int> nums = {1, 2, 3, 1};
  bool ans = Solution().containsDuplicate(nums);
  cout << ans << endl;
  return 0;
}
